crates/l2/proposer/mod.rs

```diff
@@ -108,26 +106,28 @@ impl Proposer {
                     "Failed to get block by hash from storage".to_string(),
                 ))?;
 
-            let withdrawal_txs: Vec<&Transaction> = block
+            let withdrawal_txs: Vec<H256> = block
                 .body
                 .transactions
                 .iter()
-                .filter(|tx| match tx {
-                    Transaction::PrivilegedL2Transaction(tx) => {
-                        tx.tx_type == PrivilegedTxType::Withdrawal
+                .filter_map(|tx| match tx {
+                    Transaction::PrivilegedL2Transaction(priv_tx)
+                        if priv_tx.tx_type == PrivilegedTxType::Withdrawal =>
+                    {
+                        let to = match priv_tx.to {
+                            TxKind::Call(to) => to,
+                            _ => panic!("Withdrawal transaction should be a Call"),
```

Comment:
If the `to` of `Transaction::PrivilegedL2Transaction`  is `TxKind::Create` I think it won't be very much of a trouble to ignore it (this is returning `None`, instead of panicking.

Reply:
Done [5ec12b6](https://github.com/lambdaclass/lambda_ethereum_rust/pull/961/commits/5ec12b6c6f2c8bd8a6b7ae386b3e324ee4e25a24)